net/cifsd-tools/files/cifsd.init

```diff
@@ -162,27 +167,41 @@ start_service()
 	logger -t 'cifsd' "Starting CIFS/SMB userspace service."
 	procd_open_instance
 	procd_set_param command /usr/sbin/cifsd --n
+	procd_set_param file /var/etc/cifs/smb.conf
 	procd_close_instance
 }
 
 stop_service()
 {
 	logger -t 'cifsd' "Stopping CIFSD userspace service."
 	killall cifsd > /dev/null 2>&1
-	sleep 1
-	[ -e /sys/class/cifsd-control/kill_server ] && echo hard > /sys/class/cifsd-control/kill_server
-	sleep 2
+	
+	[ -e /sys/module/cifsd ] && rmmod cifsd > /dev/null 2>&1
+	# With open smb connections rmmod is not possible, without waiting for the long 'ipc timeout', so we use 'kill_server'!
+	if [ -e /sys/module/cifsd ]; then
+		logger -t 'cifsd' "triggering kill_server"
+		if [ -e /sys/class/cifsd-control/kill_server ]; then
+			echo hard > /sys/class/cifsd-control/kill_server
+			# we need a extra timeout for the reset
+			sleep 5
+		fi
+	fi
+	# next try
 	[ -e /sys/module/cifsd ] && rmmod cifsd > /dev/null 2>&1
-	# With open smb connections rmmod takes longer
+	# check again
 	if [ -e /sys/module/cifsd ]; then
-		sleep 5
-		rmmod cifsd > /dev/null 2>&1
+		# wait more...
+		sleep 3
+	fi
+	# last try
+	[ -e /sys/module/cifsd ] && rmmod cifsd > /dev/null 2>&1
+	
+	if [ -e /sys/module/cifsd ]; then
+		logger -t 'cifsd' "module still loaded after 8s timeout"
```

Comment:
No need to enclose cifsd in ''. At least I don't think so...

Reply:
While on the topic, some of these if statements can be replaced with &&. I guess it's a matter of style though.

Reply:
If i search for examples i get all variants.... at least this works for me in my logs.